bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/MathUtils.java

```diff
@@ -63,4 +63,28 @@ public static long remainder(long numerator, long denominator) {
             }
         }
     }
+
+    public static long addExact(long num1, long num2) {
+        long r = num1 + num2;
+        if (((num1 ^ r) & (num2 ^ r)) < 0) {
+            throw  ErrorCreator.createError(BallerinaErrorReasons.NUMBER_OVERFLOW, INT_RANGE_OVERFLOW_ERROR);
+        }
+        return r;
+    }
+
+    public static long subtractExact(long num1, long num2) {
+        long r = num1 - num2;
+        if (((num1 ^ num2) & (num1 ^ r)) < 0) {
+            throw  ErrorCreator.createError(BallerinaErrorReasons.NUMBER_OVERFLOW, INT_RANGE_OVERFLOW_ERROR);
+        }
+        return r;
+    }
+
+    public static long multiplyExact(long num1, long num2) {
+        long r = num1 * num2;
+        if (num1 != 0 && r / num1 != num2) {
+            throw  ErrorCreator.createError(BallerinaErrorReasons.NUMBER_OVERFLOW, INT_RANGE_OVERFLOW_ERROR);
```

Comment:
if we are throwing the same error in multiple places we can reuse code by moving it to a new method

Reply:
I will look into this.

Reply:
When I moved throwing the error to a function
`private static void throwIntRangeOverflowError() {
        throw ErrorCreator.createError(BallerinaErrorReasons.NUMBER_OVERFLOW, INT_RANGE_OVERFLOW_ERROR);
    }`
although it worked in this implementation, the Codecov patch report says that the function call line was not covered by tests (but it is actually called to throw the error).
Now as going to `Math.addExact(), Math.subractExact(), Math.multiplyExact()` methods are better, when I write them using try catch blocks, it doesn't allow me to write 
`public static long subtractExact(long num1, long num2) {
        try {
            return Math.subtractExact(num1, num2);
        } catch (ArithmeticException e) {
            throwIntRangeOverflowError();
        }
    }`
I can neglect that by introducing another variable , But may I remove the function and use the statement in all the relevant places